src/xbot/common/controls/sensors/MockGamepad.java

```diff
@@ -1,40 +1,177 @@
 package xbot.common.controls.sensors;
 
+import xbot.common.math.XYPair;
+
 public class MockGamepad implements XGamepad {
 
-    MockJoystick mockLeftStick = new MockJoystick();
-    MockJoystick mockRightStick = new MockJoystick();
-    MockJoystick mockDpad = new MockJoystick();
-    MockJoystick mockLeftTrigger = new MockJoystick();
-    MockJoystick mockRightTrigger = new MockJoystick();
+    XYPair leftJoystickAxis;
+    XYPair rightJoystickAxis;
+    double leftTriggerAxis;
+    double rightTriggerAxis;
+    boolean leftBumper;
+    boolean rightBumper;
+    boolean AButton;
+    boolean BButton;
+    boolean XButton;
+    boolean YButton;
+    boolean backButton;
+    boolean startButton;
+    boolean leftStickButton;
+    boolean rightStickButton;
     
-    public MockGamepad() {
-        
+    public void setLeftStick(double x, double y){
+        leftJoystickAxis = new XYPair(x, y);
     }
     
-    @Override
-    public XJoystick getLeftStick() {
-        return mockLeftStick;
+    public void setRightStick(double x, double y){
+        rightJoystickAxis = new XYPair(x, y);
+    }
+    
+    public void setLeftTriggerPressed(double x){
+        leftTriggerAxis = x;
+    }
+    
+    public void setRightTriggerPressed(double x){
+        rightTriggerAxis = x;
     }
 
+    public void setLeftBumper(boolean pressed){
+        leftBumper = pressed;
+    }
+    
+    public void setRightBumper(boolean pressed){
+        rightBumper = pressed;
+    }
+    
+    public void setAButton(boolean pressed){
+        AButton = pressed;
+    }
+    
+    public void setBButton(boolean pressed){
+        BButton = pressed;
+    }
+    
+    public void setXButton(boolean pressed){
+        XButton = pressed;
+    }
+    
+    public void setYButton(boolean pressed){
+        YButton = pressed;
+    }
+    
+    public void setBackButton(boolean pressed){
+        backButton = pressed;
+    }
+    
+    public void setStartButton(boolean pressed){
+        startButton = pressed;
+    }
+    
+    public void setLeftStickButton(boolean pressed){
+        leftStickButton = pressed;
+    }
+    
+    public void setRightStickButton(boolean pressed){
+        rightStickButton = pressed;
+    }
+    
+    public XYPair getLeftStick(double x, double y){
+        return leftJoystickAxis = new XYPair(x, y);
```

Comment:
This doesn't seem right. You give it coordinates and it returns them? This seems like it should not have any parameters, and should just return the vector.